test/extended/openstack/loadbalancers.go

```diff
@@ -412,6 +417,61 @@ var _ = g.Describe("[sig-installer][Suite:openshift/openstack][lb][Serial] The O
+
+			g.By("Creating ingresscontroller for sharding the OCP in-built canary service")
+			name := "canary-sharding"
+			domain := "sharding-test.apps.shiftstack.com"
+			shardIngressCtrl, err := deployLbIngressController(oc, 10*time.Minute, name, domain, map[string]string{"ingress.openshift.io/canary": "canary_controller"})
+			defer func() {
+				err := oc.AdminOperatorClient().OperatorV1().IngressControllers(shardIngressCtrl.Namespace).Delete(context.Background(), shardIngressCtrl.Name, metav1.DeleteOptions{})
+				o.Expect(err).NotTo(o.HaveOccurred(), "error destroying ingress-controller created during the test")
+			}()
+			o.Expect(err).NotTo(o.HaveOccurred(), "new ingresscontroller did not rollout")
+
+			g.By("Checks from openshift perspective")
+			svc, err := oc.AdminKubeClient().CoreV1().Services("openshift-ingress").Get(context.Background(), "router-"+name, metav1.GetOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred(), "router-%q not found in openshift-ingress namespace", name)
+			o.Expect(svc.Spec.Type).Should(o.Equal(v1.ServiceTypeLoadBalancer), "Unexpected svc Type: %q", svc.Spec.Type)
+			loadBalancerId := svc.GetAnnotations()["loadbalancer.openstack.org/load-balancer-id"]
+			o.Expect(loadBalancerId).ShouldNot(o.BeEmpty(), "load-balancer-id annotation missing")
+			o.Expect(svc.Status.LoadBalancer.Ingress).ShouldNot(o.BeEmpty(), "svc.Status.LoadBalancer.Ingress should not be empty")
+			svcIp := svc.Status.LoadBalancer.Ingress[0].IP
+			o.Expect(svcIp).ShouldNot(o.BeEmpty(), "FIP missing on svc Status")
+			o.Expect(svc.Spec.ExternalTrafficPolicy).Should(o.Equal(v1.ServiceExternalTrafficPolicyTypeLocal),
+				"Unexpected ExternalTrafficPolicy on svc specs")
+			e2e.Logf("Service with LoadBalancerType exists with public ip %q and it is pointing to openstack loadbalancer with id %q", svcIp, loadBalancerId)
+
+			g.By("Checks from openstack perspective")
+			lb, err := octavialoadbalancers.Get(loadBalancerClient, loadBalancerId).Extract()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(lb.Provider).Should(o.Equal(strings.ToLower(lbProviderUnderTest)), "Unexpected provider in the Openstack LoadBalancer")
+			//In canary, two pools are created: ports 80 and 443. Checking all existing pools:
+			for i := 0; i < len(svc.Spec.Ports); i++ {
+				poolId := lb.Pools[i].ID
+				pool, err := pools.Get(loadBalancerClient, poolId).Extract()
+				o.Expect(err).NotTo(o.HaveOccurred())
+				o.Expect(pool.Protocol).Should(o.Equal("TCP"), "Unexpected protocol on Openstack LoadBalancer Pool: %q", pool.Name)
+			}
+
+			g.By("Test that the canary service is accessible through new ingressController")
+			route, err := oc.AdminRouteClient().RouteV1().Routes("openshift-ingress-canary").Get(context.Background(), "canary", metav1.GetOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred(), "canary route not found")
+			for i := 0; i < 100; i++ {
+				resp, err := httpsGetWithCustomLookup(route.Spec.Host, svcIp)
```

Comment:
is only testing request for port 443 enough?

Reply:
I found issues testing the connectivty to port 80 because the app is redirecting to 443 and then I hit cert issues.

We are covering multiple pools in the test "[sig-installer][Suite:openshift/openstack][lb][Serial] The Openstack platform should re-use an existing UDP Amphora LoadBalancer when new svc is created on Openshift with the proper annotation"